src/atdf/patch.rs

```diff
@@ -40,18 +44,20 @@ pub fn signals_to_port_fields(chip: &mut chip::Chip, tree: &xmltree::Element) ->
             .collect();
 
         for reg in port.registers.values_mut() {
-            if !reg.name.ends_with(name) {
-                log::error!("Register {:?} has a weird name!", reg.name);
+            if correct_reg_name(&reg.name, name) {
```

Comment:
I would argue that now, where the condition in this function is only a single line, I would put it inline directly:

```suggestion
            if reg.name.ends_with(name) || NEW_PORT_REGS.iter().any(|r| r == &reg.name) {
```